docs/api/microwave/impedance_calculator.rst

```diff
@@ -0,0 +1,163 @@
+.. _impedance_calculator:
+
+Impedance Calculator
+--------------------
+
+.. autosummary::
+   :toctree: ../_autosummary/
+   :template: module.rst
+
+   tidy3d.ImpedanceCalculator
+
+The :class:`~tidy3d.ImpedanceCalculator` computes characteristic impedance from electromagnetic field data using voltage and current path integrals. It supports three calculation methods depending on which integrals are provided:
+
+* **V and I method**: :math:`Z_0 = V / I` (when both voltage and current integrals are provided)
+* **P and V method**: :math:`Z_0 = V^2 / (2P)` (when only voltage integral is provided)
+* **P and I method**: :math:`Z_0 = 2P / I^2` (when only current integral is provided)
```

Comment:
This assumes those quantities are real? In frequency domain, there are a few conjugations applied